router/middleware/session/team.go

```diff
@@ -1,21 +1,93 @@
 package session
 
 import (
+	"github.com/drone/drone/cache"
+	"github.com/drone/drone/model"
+
+	log "github.com/Sirupsen/logrus"
 	"github.com/gin-gonic/gin"
 )
 
+func TeamPerm(c *gin.Context) *model.Perm {
+	v, ok := c.Get("teamperm")
+	if !ok {
+		return nil
+	}
+	u, ok := v.(*model.Perm)
+	if !ok {
+		return nil
+	}
+	return u
+}
+
+func SetTeamPerm() gin.HandlerFunc {
```

Comment:
I don't have any preference.  The only thing I did was copy/paste a lot of existing code and modify to suite my needs and make sure some unit testing passed so far.  😛 

Only thing I don't understand is what the following means and may not know how to do it ...

> As a side note, I think if we were to use this middleware, it would need to be registered with the handler.